src/client/src/Client.c

```diff
@@ -1403,6 +1405,8 @@ VOID viewItemRemoved(PContentView pContentView, UINT64 customData, PViewItem pVi
                 break;
 
             case STREAMING_TYPE_NEAR_REALTIME:
+                pKinesisVideoStream->diagnostics.droppedFragments++;

```

Comment:
We drop fragments in near realtime, but frames in offline mode? Can you explain why the difference between the 2 streaming modes?

Reply:
The SDK defines 3 modes of streaming

* Realtime - this is when a camera-like source produces frames which get pushed into the SDK and streamed - the main scenario.
* Offline - this is when frames are produced in IO/CPU speed into the SDK. This is a file upload like scenario. The main difference between this and Realtime mode is that this will block on "availability" (temporal and storage) while the the Realtime mode will push tail frames out. There are other nuances too in the way we process ACKs.
* Near-realtime - this mode is not implemented yet but this will operate at a fragment granularity. Similar to HLS/MPEG-DASH but for upload. The fragment(s) will be assembled and pushed in one go in a single session.